functions/wait/action/kubernetesUpgrade.go

```diff
@@ -51,14 +57,19 @@ func KubernetesUpgrade(t *testing.T, client *rancher.Client, clusterID string, m
 				return true, nil
 			}
 
-			t.Logf("Failed to instantiate wait poll.")
+			t.Logf("Waiting for cluster to upgrade kubernetes version...")
 			return false, nil
 		})
-		require.NoError(t, err)
+		require.NoError(t, wait_err)
+
+		if wait_err != nil {
+			t.Logf("Failed to instantiate kubernetes upgrade wait poll.")
+			return false, wait_err
+		}
 	}
 
 	if module == "ec2_k3s" || module == "ec2_rke2" || module == "linode_k3s" || module == "linode_rke2" {
-		err := wait.Poll(100*time.Millisecond, 30*time.Minute, func() (done bool, err error) {
+		wait_err := wait.Poll(100*time.Millisecond, 30*time.Minute, func() (done bool, err error) {
```

Comment:
same concern here. We should find a good balance in how often to poll. Personally I wouldn't want to see the logs every 100ms, or even every second since we're just waiting on an upgrade here. Maybe only log once just before polling that we will start to wait, then once again when we exit the poll if there's an error?

Reply:
Sure thing, i'll get that updated!

Reply:
updated + addressed!